lib/components/widgets/video_player/video_player_provider.dart

```diff
@@ -13,10 +13,12 @@ final videoPlayerProvider = StateNotifierProvider.autoDispose
     .family<VideoPlayerNotifier, VideoPlayerState, VideoPlayerArguments>(
   (ref, arguments) {
     final handler = ref.watch(videoPlayerHandlerProvider);
+    final mediaPreferences = ref.watch(mediaPreferencesProvider);
```

Comment:
Watching the mediaPrefences with `ref.watch` here before the `VideoPlayerNotifier` gets constructed will cause the video player to get re-created every time the mediaPreferences update - which isn't exactly what we want.

We want to check whether the video should start muted only once when we initialize the video. Changing the setting afterwards shouldn't affect the video player.

To do this you can pass down the `ref` argument to the `VideoPlayerNotifier` and use `_ref.read(mediaPreferencesProvider)` in `initialize()` to get the value.

You can check out the `downloadPathProvider` for reference. We are doing something similar there.

Reply:
Hey. Thanks for the great feedback and explanation, I understand the problem!
I added a new commit to fix it :)